src/network/peloton_server.cpp

```diff
@@ -0,0 +1,136 @@
+#include "event2/thread.h"
+#include <fstream>
+
+#include "network/peloton_server.h"
+#include "settings/settings_manager.h"
+
+namespace peloton {
+namespace network {
+
+int PelotonServer::recent_connfd = -1;
+SSL_CTX *PelotonServer::ssl_context = nullptr;
+
+PelotonServer::PelotonServer() {
+  port_ = settings::SettingsManager::GetInt(settings::SettingId::port);
+  max_connections_ = settings::SettingsManager::GetInt(settings::SettingId::max_connections);
+  private_key_file_ = settings::SettingsManager::GetString(settings::SettingId::private_key_file);
+  certificate_file_ = settings::SettingsManager::GetString(settings::SettingId::certificate_file);
+
+  // For logging purposes
+  //  event_enable_debug_mode();
+
+  //  event_set_log_callback(LogCallback);
+
+  // Commented because it's not in the libevent version we're using
+  // When we upgrade this should be uncommented
+  //  event_enable_debug_logging(EVENT_DBG_ALL);
+
+  // Ignore the broken pipe signal
+  // We don't want to exit on write when the client disconnects
+  signal(SIGPIPE, SIG_IGN);
+}
+
+PelotonServer &PelotonServer::SetupServer() {
+  // This line is critical to performance for some reason
+  evthread_use_pthreads();
+  if (settings::SettingsManager::GetString(settings::SettingId::socket_family) != "AF_INET")
+    throw ConnectionException("Unsupported socket family");
+
+  struct sockaddr_in sin;
+  PL_MEMSET(&sin, 0, sizeof(sin));
+  sin.sin_family = AF_INET;
+  sin.sin_addr.s_addr = INADDR_ANY;
+  sin.sin_port = htons(port_);
+
+  listen_fd_ = socket(AF_INET, SOCK_STREAM, 0);
+
+  if (listen_fd_ < 0) {
+    throw ConnectionException("Failed to create listen socket");
+  }
+
+  int conn_backlog = 12;
+  int reuse = 1;
+  setsockopt(listen_fd_, SOL_SOCKET, SO_REUSEADDR, &reuse, sizeof(reuse));
+
+  /* Initialize SSL listener connection */
+  SSL_load_error_strings();
+  SSL_library_init();
+
+  if ((ssl_context = SSL_CTX_new(TLSv1_server_method())) == nullptr)
+  {
+    throw ConnectionException("Error creating SSL context.");
+  }
+
+  LOG_INFO("private key file path %s", private_key_file_.c_str());
+  /*
+   * Temporarily commented to pass tests START
+  // register private key
+  if (SSL_CTX_use_PrivateKey_file(ssl_context, private_key_file_.c_str(),
+                                  SSL_FILETYPE_PEM) == 0)
+  {
+    SSL_CTX_free(ssl_context);
+    throw ConnectionException("Error associating private key.\n");
+  }
+  LOG_INFO("certificate file path %s", certificate_file_.c_str());
+  // register public key (certificate)
+  if (SSL_CTX_use_certificate_file(ssl_context, certificate_file_.c_str(),
+                                   SSL_FILETYPE_PEM) == 0)
+  {
+    SSL_CTX_free(ssl_context);
+    throw ConnectionException("Error associating certificate.\n");
+  }
+  * Temporarily commented to pass tests END
+  */
+  if (bind(listen_fd_, (struct sockaddr *) &sin, sizeof(sin)) < 0)
+  {
```

Comment:
You should probably run the formatter on this. It's definitely different than the other files. If we're going to trouble of refactoring it to a new file, we might as well run the formatter.